### ISSUE TITLE
| seisbench | WARNING | Parts of the input stream consist of fragments shorter than the number of input samples. Output might be empty.

### ISSUE THREAD

**Issue by HaofengWu20** [seisbench/seisbench#131]
Good open source software! It integrates several famous supervised learning models and large data training sets. Here, I want to directly feed my own seismic dataset, a **SU format file** including 1C data from different sensors, into the pre-trained model. However, a warning occured, I did as follow:
`import seisbench.models as sbm`
`from obspy import *` 
`# load pre-trianed model`
`eqt_model01 = sbm.EQTransformer.from_pretrained("original")`
`# load my dataset`
`st1 = read('xxxxxx.su')`
`stream = st1[0]`
`stream1 = Stream(stream)`
Here post my stream1 (after customizing `stream.trace.stats` information):
`1 Trace(s) in Stream: `
`
LOBSA.OBS09.Trace266.HHZ | 2014-07-04T00:00:00.000000Z - 2014-07-04T00:00:24.996000Z | 250.0 Hz, 6250 samples`
Continue:
`# use the annotate method to calculate the model's response to stream`
`eqt_preds31 = eqt_model01.annotate(stream1, strict=False)`
Waring occured:
`2022-10-07 18:55:15,857 | seisbench | WARNING | Parts of the input stream consist of fragments shorter than the number of input samples. Output might be empty.`
Continue:
`print(eqt_preds31)`
The warning is right:
`0 Trace(s) in Stream:`

-------
Could you give me some suggestions? Many thanks.

**Comment by yetinam** [comment]
Hey @HaofengWu20,

thank you for the positive feedback on SeisBench. Regarding your example, the issue is the length of the trace. EQTransformer takes as input 60 s long traces. For longer trace, SeisBench chunks them up and reassambles them. For shorter traces however, we do not automatically pad the data, but instead issue this warning. To solve your issue, the best approach would be to use longer waveform snippets. Other models, e.g., PhaseNet (30 s) or GPD (4 s, but much slower) also can deal with shorter input windows. Alternatively, you can try padding your waveforms with zeros to one minute length. However, this might introduce artifacts.

In addition, I saw that your data appears to be from OBS devices. While we haven't published it yet, we are currently working on a picker explicitly designed for OBS instruments. It will hopefully be release soon through SeisBench.

Please let me know if this answers your question.

**Comment by HaofengWu20** [comment]
OKay, After using GPD instead of EQTransformer, the program works. And I will try to pad my waveforms with zeros and watch out the possible artifacts. Thank you very much.

**Comment by HaofengWu20** [comment]
Hello @yetinam 

Has the literature or program on the picker , designed for OBS instruments, been published?

**Comment by ftilmann** [comment]
@HaofengWu20  To answer your (in this thread) off-topic question. We are aiming to  submit a preprint very soon (this month).

**Comment by HaofengWu20** [comment]
Thank you @ftilmann . I will continue to follow your work.
